mantis-common/src/main/java/io/mantisrx/common/compression/CompressionUtils.java

```diff
@@ -217,10 +217,19 @@ private static int getTotalByteSize(List<List<byte[]>> nestedEvents, byte[] deli
             for (int i = 0; i < line.length(); i++) {
                 if (line.charAt(i) != delimiterArray[delimiterCount]) {
                     if (delimiterCount > 0) {
-                        for (int j = 0; j < delimiterCount; ++j) {
-                            sb.append(delimiterArray[j]);
+                        boolean prefixMatch = true;
```

Comment:
This whole method is too fancy to me. I may fail your interview if the question is to fix this bug.

Could anyone tell me why this one liner is not sufficient to replace the whole method?
```
return Arrays.stream(bf.lines().collect(Collectors.joining("")).split(Pattern.quote(delimiter))).map(e -> new MantisServerSentEvent(e)).collect(Collectors.toList());
```

(of course please make that multiple lines 🤣 )

It passes all your tests (new and old).

Reply:
My guess is memory. I think this was meant to check lines as they are read in BufferedReader, so never need to pull the entire content into a String.

Reply:
I believe Calvin is correct -- performance and memory was the reason for this. Most events are pretty small but some people are sending monstrous (many megabyte +) events through Mantis. In the PlayAPI req/resp body logging use case batches can reach several gigabytes every few milliseconds. The large strings and the G1 collector don't get along very well.

Reply:
I was not aware of the large input, but that's a fair reason. I do not know if we can refactor the code in a way that is much simpler based on something like above, but we can tackle that separately I believe.

If we want to optimize memory/cpu for this, there are also some other ideas.
1. If the output strings could be large, we can let the `StringBuilder` have some initial capacity instead of letting it grow from beginning, e.g. `sb = new StringBuilder(4096);`.
2. If the output String can be too small, we can reuse the StringBuilder and just `setLength(0)` instead of `new StringBuilder()`.